homeassistant/components/mqtt/config.py

```diff
@@ -3,126 +3,21 @@
 
 import voluptuous as vol
 
-from homeassistant.const import (
-    CONF_CLIENT_ID,
-    CONF_DISCOVERY,
-    CONF_PASSWORD,
-    CONF_PORT,
-    CONF_PROTOCOL,
-    CONF_USERNAME,
-    CONF_VALUE_TEMPLATE,
-)
+from homeassistant.const import CONF_VALUE_TEMPLATE
 from homeassistant.helpers import config_validation as cv
 
 from .const import (
-    ATTR_PAYLOAD,
-    ATTR_QOS,
-    ATTR_RETAIN,
-    ATTR_TOPIC,
-    CONF_BIRTH_MESSAGE,
-    CONF_BROKER,
-    CONF_CERTIFICATE,
-    CONF_CLIENT_CERT,
-    CONF_CLIENT_KEY,
     CONF_COMMAND_TOPIC,
-    CONF_DISCOVERY_PREFIX,
     CONF_ENCODING,
-    CONF_KEEPALIVE,
     CONF_QOS,
     CONF_RETAIN,
     CONF_STATE_TOPIC,
-    CONF_TLS_INSECURE,
-    CONF_TLS_VERSION,
-    CONF_WILL_MESSAGE,
-    DEFAULT_BIRTH,
-    DEFAULT_DISCOVERY,
     DEFAULT_ENCODING,
-    DEFAULT_PREFIX,
     DEFAULT_QOS,
     DEFAULT_RETAIN,
-    DEFAULT_WILL,
-    PLATFORMS,
-    PROTOCOL_31,
-    PROTOCOL_311,
 )
 from .util import _VALID_QOS_SCHEMA, valid_publish_topic, valid_subscribe_topic
 
-DEFAULT_PORT = 1883
-DEFAULT_KEEPALIVE = 60
-DEFAULT_PROTOCOL = PROTOCOL_311
-DEFAULT_TLS_PROTOCOL = "auto"
-
-DEFAULT_VALUES = {
-    CONF_BIRTH_MESSAGE: DEFAULT_BIRTH,
-    CONF_DISCOVERY: DEFAULT_DISCOVERY,
-    CONF_PORT: DEFAULT_PORT,
-    CONF_TLS_VERSION: DEFAULT_TLS_PROTOCOL,
-    CONF_WILL_MESSAGE: DEFAULT_WILL,
-}
-
-CLIENT_KEY_AUTH_MSG = (
-    "client_key and client_cert must both be present in "
-    "the MQTT broker configuration"
-)
-
-MQTT_WILL_BIRTH_SCHEMA = vol.Schema(
-    {
-        vol.Inclusive(ATTR_TOPIC, "topic_payload"): valid_publish_topic,
-        vol.Inclusive(ATTR_PAYLOAD, "topic_payload"): cv.string,
-        vol.Optional(ATTR_QOS, default=DEFAULT_QOS): _VALID_QOS_SCHEMA,
-        vol.Optional(ATTR_RETAIN, default=DEFAULT_RETAIN): cv.boolean,
-    },
-    required=True,
-)
-
-PLATFORM_CONFIG_SCHEMA_BASE = vol.Schema(
-    {vol.Optional(platform.value): cv.ensure_list for platform in PLATFORMS}
-)
-
-CONFIG_SCHEMA_BASE = PLATFORM_CONFIG_SCHEMA_BASE.extend(
```

Comment:
Need to separate this stuff from the that what is imported by the platforms. Te removed parts move to `config_integration.py`